### ISSUE TITLE
Call to action breaks on add

### ISSUE THREAD

**Issue by smokeycam** [silverstripe/silverstripe-elemental-bannerblock#30]
**Overview**
If you attempt to create a call to action, upon saving the modal closes and the CMS elemental area disappears

So here is a screen recording (attached) using a vanilla SS4 install with only the following (composer file): 

"silverstripe/recipe-plugin": "^1.2",
"silverstripe/recipe-cms": "4.4.2@stable",
"silverstripe-themes/simple": "~3.2.0",
"silverstripe/elemental-bannerblock": "^2.0"

Same issue regardless if the element is in published or modified state.
![elemental-cta-issue](https://user-images.githubusercontent.com/1677761/65643504-005df180-e046-11e9-8624-c57ca922ac95.gif)

Attached Javascipt debug:
<img width="636" alt="Screen Shot 2019-09-26 at 10 10 48 AM" src="https://user-images.githubusercontent.com/1677761/65643521-0784ff80-e046-11e9-8b86-895dacb07ba2.png">

# Pull request
* [x] https://github.com/silverstripe/silverstripe-elemental-bannerblock/pull/51

**Comment by wilr** [comment]
Encountering the same issue with the banner block. Tested 2.x-dev and 2.0.2 of bannerblock and both fail @robbieaverill @ScopeyNZ

**Comment by ScopeyNZ** [comment]
What version of `elemental` and `admin` @wilr ?

**Comment by ScopeyNZ** [comment]
I think we have a weird constraint in our composer.json for this project.

**Comment by brynwhyman** [comment]
Noting that this is similar to https://github.com/dnadesign/silverstripe-elemental/issues/736

**Comment by wilr** [comment]
@ScopeyNZ This is using latest CWP (2.4)

https://gist.github.com/wilr/a7d69bd78dbeb25803c212d05a2608d2 for all the versions from composer-show. Basically..

    "require": {
        "cwp/cwp-recipe-core": "^2.4",
        "cwp/cwp-recipe-cms": "^2.4",
        "silverstripe/recipe-blog": "^1.4",
        "silverstripe/recipe-form-building": "1.4",
        "silverstripe/recipe-authoring-tools": "1.4",
        "silverstripe/recipe-collaboration": "^1.4",
        "silverstripe/recipe-reporting-tools": "^1.4",
        "silverstripe/recipe-content-blocks": "^2.4",
        "cwp/cwp-recipe-search": "^2.4",
        "silverstripe/recipe-services": "^1.4"
}

**Comment by Juanitou** [comment]
This is still the case after upgrading ```dnadesign/silverstripe-elemental``` to 4.3.0 (but we have at least an error displayed now).

**Comment by brynwhyman** [comment]
Hi, there's been a recent fix to the `silverstripe/versioned-admin` module which should resolve this issue, see https://github.com/silverstripe/silverstripe-versioned-admin/pull/132

Try updating to the latest dev version of versioned-admin 1.1.x or 1.2.x and let us know how you get on.

**Comment by Juanitou** [comment]
Hi. Thanks for working on this.

I have not been able to find the correct composer constraint to force the installation of the alpha version (1.3) of ```silverstripe/versioned-admin``` (now at 1.2.3) with SilverStripe 4.4.4.

As I’m in a hurry, I cannot research further now, but I’ll sure come back here to share what I’ll find.

**Comment by Juanitou** [comment]
Hi again. I’ve been able to upgrade to the latest ```silverstripe/versioned-admin``` version, which includes the fix, but the error is still there. To be precise, I’ll recap what I’m seeing (I should maybe create a new PR):

1. I cannot access the content of banner blocks created before the fix, the console shows the following error, even after saving and refreshing the CMS:
![image](https://user-images.githubusercontent.com/807463/69140598-15458700-0ac3-11ea-99f2-ac0f295aefaf.png)

2. So, this only happens with old banner blocks. I can access the content of old plain content blocks and, crucially, newly created banner blocks, even if these include an image or a call to action link.

I have checked the database trying to find a difference between old and new blocks, AND I FOUND IT!: The new banner blocks are created with the SQL `NULL` value in the `CallToActionLink` column, while the old blocks had a “null” string value.

In summary, replacing “null” by `NULL` in the `CallToActionLink` columns of the `S_EB_BannerBlock_*` tables solves the issue for me!

I hope it helps somebody.

**Comment by brynwhyman** [comment]
We're keeping this issue open to give time to look into a more solid fix for the differing `null` entries described by @Juanitou.

This regression occurred around the time 4.3 of elemental was released in September 2019, so it's relatively likely new projects created blocks between then and before [the fix](https://github.com/silverstripe/silverstripe-versioned-admin/pull/132 was released that could be affected.

I'm dropping the impact of this issue given the workaround mentioned above. I'd also presume (untested) that if block history was less important for a site, the block could be archived and recreated.

**Comment by emteknetnz** [comment]
Fixed in https://github.com/silverstripe/silverstripe-elemental-bannerblock/pull/51

**Comment by maxime-rainville** [comment]
https://github.com/silverstripe/silverstripe-elemental-bannerblock/pull/51 has been merged. I'll let you decide if this is worth adding to the sprint demo.

**Comment by emteknetnz** [comment]
The linked pull-request has been merged, I'll close this issue now
